### PR TITLE
Update Prometheus to stable 2.2.1

### PR DESCRIPTION
**Pull request by lostick**
• bump prometheus to `2.2.1`
• prometheus chart set to `0.0.21`
• kube-prometheus chart set to `0.0.31`

### CODE REVIEW

**Comment by richerve** [comment]
For helm, you will need to update prometheus Chart version too and its entry on the *requirements.yaml* file in kube-prometheus

File: Documentation/compatibility.md [hunk 1]
```diff
@@ -32,3 +32,4 @@ The versions of Prometheus compatible to be run with the Prometheus Operator are
 * v1.7.2
 * v1.8.0
 * v2.0.0
+* v2.2.0
```
Comment (brancz) [inline Documentation/compatibility.md]: This is doc is generated, it needs to be adapted in the code: https://github.com/coreos/prometheus-operator/blob/eada88b66456f7e608e450afdb80870f68746dda/pkg/prometheus/statefulset.go#L74

**Comment by gianrubio** [comment]
Should we wait to merge this before this issue be enlightened ?
https://github.com/prometheus/prometheus/issues/3943

File: Documentation/compatibility.md (hunk 1 above)
Comment (lostick) [reply Documentation/compatibility.md]: @brancz well spotted

**Comment by lostick** [comment]
@gianrubio agreed we should hold off until we get more details on this.
While I have not been able to reproduce it, it looks like it's affecting a few people.

**Comment by brancz** [comment]
I think we should just move forward, it's unclear whether the problem was already present for the rc.1 which we currently have. At this point I prefer having lots of people report to report and describe their problem than keeping the release candidate. Sadly the 2.1 release was also broken in some ways, we're actively trying to improve the situation upstream. Sorry for the trouble.

**Comment by gianrubio** [comment]
Agreed @brancz, thanks for clarifying. The CI is failing but the helm changes are good for me!!

**Comment by brancz** [comment]
can you just run `make generate` to regenerate the auto generated contents? then this lgtm

**Comment by lostick** [comment]
@brancz sure thing

**Comment by brancz** [comment]
two things: 

* 2.2.1 was just released (builds still running) but we should probably just bump that once built
* there is a problem with the helm build (I think the version needs to be bumped, but @gianrubio knows best)

**Comment by gianrubio** [comment]
rebasing this branch + following [the guide](https://github.com/coreos/prometheus-operator/tree/master/helm#how-to-contribue) should be enough :)

**Comment by brancz** [comment]
Lgtm, thanks!

**Comment by lostick** [comment]
Tests passing locally with the new charts and prometheus `2.2.1`.

**Comment by brancz** [comment]
thanks!

**Pull request merged**